src/common/LICENSE

```diff
@@ -0,0 +1,21 @@
+MIT License
```

Comment:
We list third-party code & licenses in the README.md [License section](https://github.com/PQCMayo/MAYO-C?tab=readme-ov-file#license). Better move the notice there since this LICENSE file seems to only apply to the NEON AES-code.